embassy-sync/src/pipe.rs

```diff
@@ -293,7 +293,19 @@ where
     pub fn write<'a>(&'a self, buf: &'a [u8]) -> WriteFuture<'a, M, N> {
         WriteFuture { pipe: self, buf }
     }
-
+    
+    /// Write all bytes to the pipe.
+    ///
+    /// This method writes all bytes from `buf` into the pipe
+    pub async fn write_all<'a>(&'a self, buf: &'a [u8]) -> Result<(), TryWriteError> {
+        let mut buf = buf;
```

Comment:
intead of creating another `let` binding, you can make the function arg itself mut: `fn write_all(..., mut buf: &[u8])`